packages/contracts-ops/contracts/scripts/InitializeImplementations.s.sol

```diff
@@ -2,40 +2,77 @@
 pragma solidity 0.7.6;
 pragma abicoder v2;
 
-import {DeployImplementations} from "./DeployImplementations.s.sol";
-import {IUpdaterManager} from "@nomad-xyz/contracts-core/contracts/interfaces/IUpdaterManager.sol";
+import "forge-std/Script.sol";
+import {Config} from "../Config.sol";
+import {DeployImplementationsLogic} from "./DeployImplementations.s.sol";
 
-contract DeployAndInitializeImplementations is DeployImplementations {
-    // entrypoint
-    function deployAndInitialize(
-        string memory _configPath,
-        string memory _domain
-    ) public {
-        // deploy implementations & write their values to _outputFile
-        deploy(_configPath, _domain);
-        // initialize implementations
-        initializeImplementations(_domain);
-    }
+// Core
+import {Home} from "@nomad-xyz/contracts-core/contracts/Home.sol";
+import {Replica} from "@nomad-xyz/contracts-core/contracts/Replica.sol";
+import {GovernanceRouter} from "@nomad-xyz/contracts-core/contracts/governance/GovernanceRouter.sol";
+import {IUpdaterManager} from "@nomad-xyz/contracts-core/contracts/interfaces/IUpdaterManager.sol";
+// Bridge
+import {BridgeRouter, EthereumBridgeRouter} from "@nomad-xyz/contracts-bridge/contracts/BridgeRouter.sol";
+import {BridgeToken} from "@nomad-xyz/contracts-bridge/contracts/BridgeToken.sol";
+import {TokenRegistry} from "@nomad-xyz/contracts-bridge/contracts/TokenRegistry.sol";
 
+contract InitializeImplementationsLogic is Script, Config {
     // NOTE: init values are zero wherever possible
     // Storage variables in implementation contracts don't matter.
     function initializeImplementations(string memory _domain) internal {
         // Home
         // NOTE: must pass real value for UpdaterManager because it is queried
+        Home home = Home(homeUpgrade(_domain).implementation);
         home.initialize(IUpdaterManager(address(getUpdaterManager(_domain))));
```

Comment:
add a check that not already initialized. skip if so

Reply:
e5c8c961

Reply:
https://github.com/nomad-xyz/monorepo/pull/511/commits/e5c8c961714fa8362837fe62c3e2530a54fd7503